### DISCUSSION TITLE
How to disable public sign-up?

### DISCUSSION THREAD

**Discussion by salim-b** [formbricks/formbricks#112] [General Requests] [answered]
Thanks for this promising open source project!

I wonder: Is it possible to configure snoopForms to have public sign-up disabled? And does it instead allow existing users to *invite* new users? Or how would I control user account creation on a self-hosted instance?

**Comment by mattinannt** [comment]
@salim-b currently it's not possible but we are working on it.
in the long term creating a default user and then letting this user invite new users would be the best way I think.

the easiest fix we can build within the next days would be to disable signups via an environment variable. But then you need to rebuild the code in order to disable signup again, but I think it's a start and then we can build the invite feature and default user.

**Comment by mattinannt** [comment, marked as answer]
@salim-b We now added an option to disable signup via environment variable ( https://github.com/formbricks/snoopforms/commit/a4c571ffd3766cf88068fa428cf7f224d05f5761 ). You can set `NEXT_PUBLIC_SIGNUP_DISABLED=1` to disable the signup functionality. You need to rebuild the snoopForms code after you changed the environment variable for the change to take effect.

In the future we will also build an option to create an initial user on snoopForms setup and an option to invite users when signup is disabled but this is a fast workaround that enables you to secure your instance.

**Reply by salim-b** [reply]
Cool, thanks a lot for adding the `NEXT_PUBLIC_SIGNUP_DISABLED` env var, works!

> In the future we will also build an option to create an initial user on snoopForms setup and an option to invite users when signup is disabled

That would be perfect, I guess. Looking forward to it!
